adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -0,0 +1,152 @@
+package software.amazon.adot.testbed;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.MountableFile;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsRequest;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Testcontainers(disabledWithoutDocker = true)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class LogsTests {
+    private static final String TEST_IMAGE = System.getenv("TEST_IMAGE") != null && !System.getenv("TEST_IMAGE").isEmpty()
+        ? System.getenv("TEST_IMAGE")
+        : "public.ecr.aws/aws-observability/aws-otel-collector:latest";
+    private final Logger collectorLogger = LoggerFactory.getLogger("collector");
+    private static final String uniqueID = UUID.randomUUID().toString();
+
+    private GenericContainer<?> collector;
+
+    LogsTests() throws Exception {
+    }
+
+    private GenericContainer<?> createAndStartCollector(String configFilePath, String logFilePath, String uniqueLogStreamName) throws IOException {
```

Comment:
Consider renaming `uniqueLogStreamName`. Log streams must be unique anyways as a [requirement](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_CreateLogStream.html). This also doesn't provide any context on why you would want to use unique log streams for these tests. `createAndStartCollector` is and IMO should be generic. Log stream uniqueness for the test case is a constraint for our tests to execute properly.

Reply:
Renamed the usage